aiocache/backends/redis.py

```diff
@@ -52,163 +39,171 @@ def __init__(
         port=6379,
         db=0,
         password=None,
-        pool_min_size=1,
-        pool_max_size=10,
+        pool_min_size=_NOT_SET,
+        pool_max_size=None,
         loop=None,
         create_connection_timeout=None,
         **kwargs,
     ):
         super().__init__(**kwargs)
+        if loop is not None:
+            warnings.warn(
+                "Parameter 'loop' has been obsolete on aiocache >= 0.12.0",
+                DeprecationWarning,
+            )
+        if pool_min_size is not _NOT_SET:
+            warnings.warn(
+                "Parameter 'pool_min_size' has been obsolete since aiocache >= 0.12.0",
+                DeprecationWarning,
+            )
+
         self.endpoint = endpoint
         self.port = int(port)
         self.db = int(db)
         self.password = password
-        self.pool_min_size = int(pool_min_size)
-        self.pool_max_size = int(pool_max_size)
+        # NOTE: In redis and aioredis 2.x, ConnectionPool raises ConnectionError
+        #  but doesn't wait for connection reuse when number of conns
+        #  exceeds the limit. While in aioredis 1.x, it waits asynchronously
+        #  via asyncio.Condition. So max pool size defaults to None in redis.
+        if pool_max_size is None:
+            self.pool_max_size = None
+        else:
+            self.pool_max_size = int(pool_max_size)
         self.create_connection_timeout = (
             float(create_connection_timeout) if create_connection_timeout else None
         )
-        self.__pool_lock = None
-        self._loop = loop
-        self._pool = None
-
-    @property
-    def _pool_lock(self):
-        if self.__pool_lock is None:
-            self.__pool_lock = asyncio.Lock()
-        return self.__pool_lock
-
-    async def acquire_conn(self):
-        await self._get_pool()
-        conn = await self._pool.acquire()
-        conn = aioredis.Redis(conn)
-        return conn
-
-    async def release_conn(self, _conn):
-        self._pool.release(_conn)
-
-    @conn
+
+        redis_kwargs = {
+            "host": self.endpoint,
+            "port": self.port,
+            "db": self.db,
+            "password": self.password,
+            # NOTE: decoding can't be controlled on API level after switching to
+            #  redis, we need to disable decoding on global/connection level.
+            #  Cause some of the values are save as bytes directly, like pickle
+            #  serialized values, which may raise exc when decoded with 'utf-8'.
+            "decode_responses": False,
+            # parameter names are different
+            "socket_connect_timeout": self.create_connection_timeout,
+            "max_connections": self.pool_max_size,
+        }
+        self.client = redis.Redis(**redis_kwargs)
```

Comment:
Better to put the kwargs in directly. You lose typing information with a dict, so mypy is unlikely to catch wrong types getting passed in (e.g. if upstream API changes).